lua/apple-music/init.lua

```diff
@@ -44,6 +44,11 @@ local execute = function(cmd)
 	return pcall(exe, cmd)
 end
 
+--- Escape single and double quotes from names
+local sanitize_name = function(name)
+	return name:gsub("'", "'\\''"):gsub('"', '\\"')
```

Comment:
I still found an edge case that this change does not handle. When the song has an escape character (`\`) before a single or double quote it still breaks it. I've found the following song for this:

<img width="632" alt="Screenshot 2024-07-27 at 08 38 26" src="https://github.com/user-attachments/assets/6feeace5-3729-45f8-b951-ad3bb252bd8b">

Or we can just create a playlist that has these types of magic characters (not that I'd imagine anyone doing this):

<img width="195" alt="Screenshot 2024-07-27 at 08 39 54" src="https://github.com/user-attachments/assets/45a5824d-d82e-4f25-906a-22ff6ef8d7d2">

For this, to be solved I'd suggest adding an extra gsub before the other ones, e.g.:

```
diff --git a/lua/apple-music/init.lua b/lua/apple-music/init.lua
index e8e4bf4..6dca0e3 100644
--- a/lua/apple-music/init.lua
+++ b/lua/apple-music/init.lua
@@ -46,7 +46,7 @@ end
 
 --- Escape single and double quotes from names
 local sanitize_name = function(name)
-	return name:gsub("'", "'\\''"):gsub('"', '\\"')
+	return name:gsub("\\", "\\\\"):gsub("'", "'\\''"):gsub('"', '\\"')
 end
 
 local grab_major_os_version = function()
```

Reply:
Haha 😆 !! From the album "Escape Sequence" really!? This is actually really funny to me. It's tempting to punish the artist for choosing such a cruel name.

Thanks for finding this. 👍 

An alternate approach I found was using `quoted form of` from applescript but I didn't really wanna mess around with it for too long. I'll apply this and test it later today and put it out tonight. ✅

Reply:
Yeah, I think it's a good enough workaround, I was also thinking of something like this, not sure if there are better ways to do this though. LGTM 🫡